client/core/types.go

```diff
@@ -27,29 +27,47 @@ type WalletForm struct {
 	INIPath string
 }
 
-// WalletStatus is the current status of an exchange wallet.
-type WalletStatus struct {
+// WalletState is the current status of an exchange wallet.
+type WalletState struct {
 	Symbol  string `json:"symbol"`
 	AssetID uint32 `json:"assetID"`
 	Open    bool   `json:"open"`
 	Running bool   `json:"running"`
+	Updated uint64 `json:"updated"`
+	Balance uint64 `json:"balance"`
+	Address string `json:"address"`
+	FeeRate uint64 `json:"feerate"`
+	Units   string `json:"units"`
 }
 
 // User is information about the user's wallets and DEX accounts.
 type User struct {
-	Markets     map[string][]*Market `json:"markets"`
-	Wallets     []*WalletStatus      `json:"wallets"`
-	Initialized bool                 `json:"inited"`
+	Markets map[string][]*Market `json:"markets"`
+	// Wallets     []*WalletState       `json:"wallets"`
+	Initialized bool                       `json:"inited"`
+	Assets      map[uint32]*SupportedAsset `json:"assets"`
+}
+
+// SupportedAsset is data about an asset and possibly the wallet associated
+// with it.
+type SupportedAsset struct {
+	ID     uint32            `json:"id"`
+	Symbol string            `json:"symbol"`
+	Wallet *WalletState      `json:"wallet"`
+	Info   *asset.WalletInfo `json:"info"`
 }
 
 // xcWallet is a wallet.
 type xcWallet struct {
 	asset.Wallet
-	waiter   *dex.StartStopWaiter
-	AssetID  uint32
-	mtx      sync.RWMutex
-	lockTime time.Time
-	hookedUp bool
+	connector *dex.ConnectionMaster
+	AssetID   uint32
+	mtx       sync.RWMutex
+	lockTime  time.Time
+	hookedUp  bool
+	balance   uint64
+	updated   time.Time
```

Comment:
Does `updated` refer only to `balance`?

Reply:
It does refer only to the balance. Will rename to reflect that. 

It is updated in via `setBalance` in `CreateWallet` and `connectedWallet` right now. I'm assuming the information will be useful both internally and in the UI eventually, but it is not currently used for anything.

Reply:
Sounds good, just making sure.